### ISSUE TITLE
iceland_ð in script stops prompt to install missing R packages from appearing

### ISSUE THREAD

**Issue by JMSteinberg** [rstudio/rstudio#10512]
### System details

    RStudio Edition : Desktop
    RStudio Version :  2022.02.0 Build 392 - and previous
    OS Version      :  x86_64-w64-mingw32/x64 (64-bit)
    R Version       : 4.1.2

### Steps to reproduce the problem
if there is a package that is not installed in the script, and there is a non-quoted Icelandic character (ð) in the script, upon saving, the prompt to install the package will not appear
### Describe the problem in detail
the gt package is not installed on my mac running Windows 10.  Either in 1252 or UTF-8 the prompt does not appear upon saving this script:
library(gt)
site <- c("Garður", "Gunnlaugsgerði", "Keflavík")
iceland_ð <- c(1, 2, 3)
data.frame(site, iceland_ð)

### Describe the behavior you expected
with this script, the prompt to install the gt package appears:

library(gt)
site <- c("Garður", "Gunnlaugsgerði", "Keflavík")
iceland_d <- c(1, 2, 3)
data.frame(site, iceland_d)

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by JMSteinberg** [comment]
You are right,  I can't make it happen on a Mac.  

On  Windows 10 with 2022.02.0+420 the prompt disappears with an Icelandic ð (I have never downloaded a daily build before).  The same behavior also happens on my surface.

This is a screenshot with normal d:
![Win_420](https://user-images.githubusercontent.com/63022391/152855755-1b9b0912-abd4-4fe6-a3bb-87998c0e9eeb.JPG)
and after changing to a special Icelandic d (ð) and saving the script, the prompt to install the missing package disappears:
![Win_420_Ice](https://user-images.githubusercontent.com/63022391/152856010-af777d85-ec21-440f-b416-1b249d300e67.JPG)
Can you make this happen?

**Comment by kevinushey** [comment]
I was able to reproduce (R 4.1.3 on Windows). The problem here is that our attempt to parse the code fails, with this error:

```
<simpleError in parse(text = code, encoding = "UTF-8"): <text>:4:9: unexpected input
3: site <- c("GarÃ°ur", "GunnlaugsgerÃ°i", "KeflavÃ­k")
4: iceland_Ã°
           ^>
```

For example:

```
code <- enc2utf8('
library(gt)
site <- c("Garður", "Gunnlaugsgerði", "Keflavík")
iceland_ð <- c(1, 2, 3)
data.frame(site, iceland_ð)
')

parse(text = code, encoding = "UTF-8")
```

Somewhat surprisingly, this does work if we _don't_ specify the encoding:

```
> code <- enc2utf8('
+ library(gt)
+ site <- c("Garður", "Gunnlaugsgerði", "Keflavík")
+ iceland_ð <- c(1, 2, 3)
+ data.frame(site, iceland_ð)
+ ')
> parse(text = code)
expression(library(gt), site <- c("Garður", "Gunnlaugsgerði", "Keflaví, 
    iceland_ð <- c(1, 2, 3, data.frame(site, iceland_ð)
```

but presumedly this is because the text is then re-encoded as latin1 and can be successfully parsed there.

FWIW, I expect this issue to be resolved with R 4.2.0 so ultimately you might want to just wait and upgrade then.
